File: cs50 assignments/mario.c
// 1st Assignment. mario.c prints pyramids to mimic the Mario game environment. For specifications, view: https://cs50.harvard.edu/x/2022/psets/1/mario/more/


#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // asks for integer 1-8
    int x;
    do
    {
        x = get_int("Height: ");
    }
    while (x < 1 || x > 8);

    // counter for #
    int y = 1;
    // counter for spaces
    int z = x - 1;
    do
    {
        // prints left-side spaces
        for (int j = 0; j < z ; j++)
        {
            printf(" ");
        }
        // prints left-side #
        for (int i = 0; i < y; i++)
        {
            printf("#");
        }
        // prints center spaces
        printf("  ");
        // prints right side #
        for (int i = 0; i < y; i++)
        {
            printf("#");
        }
        printf("\n");
        y++;
        z--;
    }
    while (y <= x);
}

File: cs50 assignments/mario.py
#13th Assignment. mario.py is a reimplementation of mario.c in python.

# TODO
from cs50 import get_int


def main():

    # prompt user for height and reprompt if value is not 1-8
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break

    # call function to draw hashes
    draw(height, height)


def draw(n, h):

    # stop loop if n = 0
    if n <= 0:
        return

    # recursively draw hashes starting from 1
    draw(n - 1, h)

    # draw left-side spaces
    for i in range(h - n):
        print(" ", end="")

    # draw left-side hashes
    for i in range(n):
        print("#", end="")

    # draw center spaces
    print("  ", end="")

    # draw right-side hashes
    for i in range(n):
        print("#", end="")

    # start a new line with each loop
    print("")


# run main function
main()
